node/lib/Munin/Node/SpoolReader.pm

```diff
@@ -96,22 +104,18 @@ sub list
 }
 
 
+# The third argument ($submission_function) is optional: it is used for processing partial results,
+# in order to avoid slurping the full data content into RAM.  The result is empty, if this function
+# is given.
 sub _cat_multigraph_file
 {
-    my ($self, $service, $timestamp, $max_samples_per_service) = @_;
-
-    # Default $max_samples_per_service is 5, in order to have a 5x time
-    # increase in catchup.  This enables to not overwhelm the munin-update when
-    # there is big backlog to handle. Use "0" to send an infinite number of
-    # samples
-    $max_samples_per_service = 5 if (! defined $max_samples_per_service);
```

Comment:
I also tried with the max_samples_per_service, but it seems it isn't doing anything. But if we can get it working, it would help with the timeout of a very big transfer.

Reply:
Going to test smaller samples than the default. Rereading this loop with the max_samples_per_service, I probably will get it working with setting the samples to just 1h instead of 1 day and increasing the retain on munin-asyncd

Reply:
> I also tried with the max_samples_per_service, but it seems it isn't doing anything.

Do you think, its implementation needs to be fixed?

> But if we can get it working, it would help with the timeout of a very big transfer.

OK, so we will keep that commit.

Reply:
the `$max_samples_per_service` should indeed limit the amount of things we sent in one lump. If it isn't really doing anything, there might be a bug indeed.

Reply:
@steveschnepp the `$max_samples_per_service` is working as it should, hence the addition of `sort` was needed to get the right order of the samples. And a smaller sample size on munin-asyncd was needed, or a timeout occured. I was just confused by the result.